### PR TITLE
Javascript week2 Audrey

### PR DESCRIPTION
**Pull request by gaechkahub**
All exercises of the homework.

### CODE REVIEW

File: 1-JavaScript/Week2/homework/ex6-totalCost.js [hunk 1]
```diff
@@ -21,22 +21,33 @@ instead!
...
 };
 
-function calculateTotalPrice(/* TODO parameter(s) go here */) {
-  // TODO replace this comment with your code
+function calculateTotalPrice(object) {
+  if (!object) {
+    return 'Function need one parameter';
+  }
+  const amount = Object.values(object).reduce((a, b) => a + b, 0);
+  return `Total: €${amount}`;
```
Comment (dardecena) [inline 1-JavaScript/Week2/homework/ex6-totalCost.js:36]: 👍 
Fantastic job using the reduce method. To further improve your answer, could you refactor it in a way that the return amount is in this format € x . xx (to the hundreths place).

File: 1-JavaScript/Week2/homework/ex6-totalCost.js [hunk 2]
```diff
@@ -21,22 +21,33 @@ instead!
...
 }
+console.log(calculateTotalPrice(cartForParty));
 
 // ! Test functions (plain vanilla JavaScript)
 function test1() {
   console.log('\nTest 1: calculateTotalPrice should take one parameter');
-  // TODO replace this comment with your code
+  console.assert(calculateTotalPrice.length === 1);
```
Comment (dardecena) [inline 1-JavaScript/Week2/homework/ex6-totalCost.js:43]: 👍

File: 1-JavaScript/Week2/homework/ex6-totalCost.js [hunk 3]
```diff
@@ -21,22 +21,33 @@ instead!
...
 }
 
 function test2() {
   console.log('\nTest 2: return correct output when passed cartForParty');
-  // TODO replace this comment with your code
+  const expected = 'Total: €11.7';
+  const actual = calculateTotalPrice(cartForParty);
+  console.assert(actual === expected);
```
Comment (dardecena) [inline 1-JavaScript/Week2/homework/ex6-totalCost.js:50]: 👍

File: 1-JavaScript/Week2/homework/ex7-mindPrivacy.js [hunk 4]
```diff
@@ -30,8 +30,17 @@ const employeeRecords = [
...
+  if (!array) {
+    return 'Error: for this function an array is needed';
+  }
+  const filteredRecords = array.map(({ name, occupation, email }) => ({
+    name,
+    occupation,
+    email,
+  }));
+  console.log(filteredRecords);
+  return filteredRecords;
```
Comment (dardecena) [inline 1-JavaScript/Week2/homework/ex7-mindPrivacy.js:43]: 👍 
I like how you start off with a data type check. You demo a good understanding of using the map method.

**Review by dardecena — APPROVED** [review]
You did a wonderful job Audrey! 

I added a few optional challenges for you, but you are not obligated to complete them if you prefer not to.

File: 1-JavaScript/Week2/homework/ex5-shoppingCartPure.js [hunk 5]
```diff
@@ -16,8 +16,15 @@ it pure. Do the following:
...
-  // TODO complete this function
+function addToShoppingCart(listOfItems, item) {
+  if (!listOfItems || !item) {
+    return 'You need to provide two arguments to execute this function.';
+  }
+  const newList = [...listOfItems, item];
+  if (newList.length > 3) {
+    newList.shift();
+  }
+  return newList;
```
[the remark this answers is not in the document]
Comment (gaechkahub) [reply 1-JavaScript/Week2/homework/ex5-shoppingCartPure.js:27]: > There's a way to implement this function with two lines of code using the spread syntax. Could you propose a solution?

function addToShoppingCart(listOfItems, item) {
  return !listOfItems || !item
    ? 'You need to provide two arguments to execute this function.'
    : [...listOfItems.slice(-2), item];
 }

File: 1-JavaScript/Week2/homework/ex6-totalCost.js (hunk 1 above)
Comment (gaechkahub) [reply 1-JavaScript/Week2/homework/ex6-totalCost.js:36]: > 👍 Fantastic job using the reduce method. To further improve your answer, could you refactor it in a way that the return amount is in this format € x . xx (to the hundreths place).

return `Total: €${amount.toFixed(2)}`;

**Pull request closed without merging**